include/fastdds/rtps/transport/shared_mem/SharedMemTransportDescriptor.h

```diff
@@ -56,30 +56,22 @@ typedef struct SharedMemTransportDescriptor : public TransportDescriptorInterfac
     {
         return segment_size_;
     }
-
-    /**
-     * Sets the segment_size and the max_message_size.
-     * max_message must be <= segment_size
-     * @param [in] segment_size in bytes.
-     * @param [in] max_message_size in bytes.
-     */
+    
     RTPS_DllAPI void segment_size(
-            uint32_t segment_size,
-            uint32_t max_message_size)
+            uint32_t segment_size)
     {
         segment_size_ = segment_size;
-        maxMessageSize = max_message_size;
     }
 
-    RTPS_DllAPI OverflowPolicy port_overflow_policy() const
-    {
-        return port_overflow_policy_;
+    virtual uint32_t max_message_size() const override
+    { 
+        return maxMessageSize; 
     }
 
```

Comment:
This override is doing the same as the parent. I think we can remove this override
```suggestion
```

Reply:
It doesn't compile doing that. I will investigate that later